ptyprocess/ptyprocess.py

```diff
@@ -699,10 +699,25 @@ def isalive(self):
         except OSError as e:
             # No child processes
             if e.errno == errno.ECHILD:
-                raise PtyProcessError('isalive() encountered condition ' +
-                        'where "terminated" is 0, but there was no child ' +
-                        'process. Did someone else call waitpid() ' +
-                        'on our process?')
+                try:
+                    os.kill(self.pid, 0)
+                except OSError as err:
+                    if err.errno == errno.ESRCH:
+                        # No such process
+                        raise PtyProcessError('isalive() encountered condition ' +
+                                'where "terminated" is 0, but there was no child ' +
+                                'process. Did someone else call waitpid() ' +
+                                'on our process?')
+                    elif err.errno == errno.EPERM:
+                        # Process deny access
+                        raise PtyProcessError('isalive() encountered condition ' +
+                                'where the child process deny access')
```

Comment:
Maybe

    raise PtyProcessError('isalive() does not have permission to query the process')